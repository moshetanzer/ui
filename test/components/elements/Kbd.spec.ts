import { describe, it, expect } from 'vitest'
import { UKbd } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Kbd', () => {
  it.each([
    ['with value', { props: { value: 'O' } }],
    ['with custom size', { props: { value: 'O', size: 'xs' } }],
    ['with default slot', { slots: { default: () => 'Default slot' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UKbd.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UKbd)
    expect(html).toMatchSnapshot()
  })
})
