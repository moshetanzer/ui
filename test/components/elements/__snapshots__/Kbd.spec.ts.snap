// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Kbd > renders with custom size correctly 1`] = `"<kbd class="inline-flex items-center justify-center text-gray-900 dark:text-white h-4 min-w-[16px] text-[10px] px-1 rounded font-medium font-sans bg-gray-100 dark:bg-gray-800 ring-1 ring-gray-300 dark:ring-gray-700 ring-inset">O</kbd>"`;

exports[`Kbd > renders with default slot correctly 1`] = `"<kbd class="inline-flex items-center justify-center text-gray-900 dark:text-white h-5 min-w-[20px] text-[11px] px-1 rounded font-medium font-sans bg-gray-100 dark:bg-gray-800 ring-1 ring-gray-300 dark:ring-gray-700 ring-inset">Default slot</kbd>"`;

exports[`Kbd > renders with value correctly 1`] = `"<kbd class="inline-flex items-center justify-center text-gray-900 dark:text-white h-5 min-w-[20px] text-[11px] px-1 rounded font-medium font-sans bg-gray-100 dark:bg-gray-800 ring-1 ring-gray-300 dark:ring-gray-700 ring-inset">O</kbd>"`;
